src/Fixer/Phpdoc/PhpdocParamOrderFixer.php

```diff
@@ -0,0 +1,174 @@
+            $docBlock = $this->sortDocBlockParamAnnotations($tokens[$index], $funcParamTokens);
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $docBlock->getContent()]);
+        }
+    }
+
+    /**
+     * @param Tokens            $tokens
+     * @param array<int, Token> $functionSequence
+     *
+     * @return array
+     */
+    private function getFuncParameterTokens(Tokens $tokens, array $functionSequence)
+    {
+        $paramBlockStart = array_keys($functionSequence)[2];
+        $paramBlockEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $paramBlockStart);
+
+        $params = [];
+        for (
+            $i = $tokens->getNextTokenOfKind($paramBlockStart, [[T_VARIABLE]]);
+            null !== $i && $i < $paramBlockEnd;
+            $i = $tokens->getNextTokenOfKind($i, [[T_VARIABLE]])
+        ) {
+            $params[] = $tokens[$i];
+        }
+
+        return $params;
+    }
+
+    /**
+     * @param Token   $docBlockToken
+     * @param Token[] $funcParamTokens
+     *
+     * @return DocBlock
+     */
+    private function sortDocBlockParamAnnotations(Token $docBlockToken, array $funcParamTokens)
+    {
+        $doc = new DocBlock($docBlockToken->getContent());
+        $paramAnotations = $doc->getAnnotationsOfType('param');
+
+        // Create a list of valid @param annotations
+        $validParams = [];
+        foreach ($funcParamTokens as $funcParam) {
+            $index = $this->findParamAnnotationByIdentifier($paramAnotations, $funcParam->getContent());
+            if (is_int($index)) {
+                $validParams[$index] = $paramAnotations[$index]->getContent();
+            }
+        }
+
+        // Detect superfluous annotations
+        /** @var Annotation[] $invalidParams */
+        $invalidParams = array_diff_key($paramAnotations, $validParams);
+
+        // Reset indices
+        $validParams = array_values($validParams);
+        $invalidParams = array_values($invalidParams);
+
+        // Get annotation content and fix index
+        foreach ($invalidParams as $i => $params) {
+            unset($invalidParams[$i]);
+            $invalidParams[$i + count($validParams)] = $params->getContent();
+        }
+
+        // Concatenate the sorted annotations
+        $orderedAnnotations = $validParams + $invalidParams;
+
+        // Rewrite the param annotations
+        foreach ($paramAnotations as $i => $docAnnotation) {
+            $docAnnotation->remove();
+            $doc
+                ->getLine($docAnnotation->getStart())
+                ->setContent($orderedAnnotations[$i]);
+        }
+
+        return $doc;
+    }
+
+    /**
+     * @param Annotation[] $docParams
+     * @param string       $identifier
+     *
+     * @return null|int
+     */
+    private function findParamAnnotationByIdentifier(array $docParams, $identifier)
+    {
+        $regex = sprintf('/\*\s*@param\s*\w*\s*\%s/', $identifier);
```

Comment:
It would be cleaner to use `preg_quote` instead of escaping `%s`.

Reply:
Good call. The regex has also changed to support more complex types like `Foo[]|\Bar\Baz`. It doesn't check the order of operators, though. I'm not sure if it's necessary to take this into account. Invalid `@param` annotations would be possible right now, but I don't know if it should be the fixer's task to check validity thoroughly.